tensorboard/plugins/mesh/mesh_plugin.py

```diff
@@ -194,8 +200,12 @@ def _collect_tensor_events(self, request):
       meta = self._instance_tag_to_metadata[instance_tag]
       tensor_events += [(meta, tensor) for tensor in tensors]
 
-    # Make sure tensors sorted by timestamp in ascending order.
-    tensor_events = sorted(
+    if timestamp is not None:
+      tensor_events = [
+        event for event in tensor_events if event[1].wall_time == timestamp]
```

Comment:
Why are we filtering for timestamp equality here? The conventional way
to select a single step is to filter by step, not timestamp. Especially
given that mesh data is emitted as up to three separate summaries, it
seems plausible that the different components for a single step might
have been written out at different times.

Reply:
I thought the timestamp applies to summaries at the same step, no? Otherwise how do you organize summaries in steps?

Reply:
Not sure what you mean: `wall_time` and `step` are [two separate fields
on a proto that’s written to disk][1]; we don’t backfill steps at
runtime by grouping timestamps.

We use `step` as a key rather than `wall_time` because (among other
things) it’s an integer rather than an arbitrary floating-point number.

[1]: https://github.com/tensorflow/tensorboard/blob/master/tensorboard/compat/proto/event.proto#L14-L19